src/logic/operate.js

```diff
@@ -0,0 +1,35 @@
+import Big from 'big.js';
+
+const operate = (numberOne, numberTwo, operation) => {
+  let result;
+  try {
+    const num1 = Big(+numberOne);
+    const num2 = Big(+numberTwo);
+
+    if (operation === '+') {
+      result = Big(num1.plus(num2));
+    }
+
+    if (operation === '-') {
+      result = Big(num1.minus(num2));
+    }
+
+    if (operation === 'X') {
+      result = Big(num1.times(num2));
+    }
+
+    if (operation === '÷') {
+      result = Big(num1.div(num2));
+    }
+
+    if (operation === '%') {
```

Comment:
- [ ] `%` in our case means turning a number into a percentage.

![](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone4/performOperate.png)

Reply:
Made the change you wanted.